native/src/main/java/io/ballerina/stdlib/serdes/BallerinaStructuredTypeMessageSerializer.java

```diff
@@ -0,0 +1,136 @@
+                break;
+            default:
+                throw createSerdesError(UNSUPPORTED_DATA_TYPE + type.getName(), SERDES_ERROR);
+        }
+    }
+
+    public MessageSerializer getMessageSerializer() {
+        return messageSerializer;
+    }
+
+    public void setMessageSerializer(MessageSerializer messageSerializer) {
+        this.messageSerializer = messageSerializer;
+    }
+
+    public Builder generateMessage() {
+        List<MessageFieldData> fieldNamesAndValues = messageSerializer.getListOfMessageFieldData();
+
+        for (MessageFieldData entry : fieldNamesAndValues) {
+            Object ballerinaValue = entry.getBallerinaValue();
+            Type ballerinaType = entry.getBallerinaType();
+            String fieldName = entry.getFieldName();
+
+            messageSerializer.setCurrentFieldName(fieldName);
+
+            switch (ballerinaType.getTag()) {
+                case TypeTags.NULL_TAG:
+                    messageSerializer.setNullFieldValue(ballerinaValue);
+                    break;
+                case TypeTags.INT_TAG:
+                    messageSerializer.setIntFieldValue(ballerinaValue);
+                    break;
+                case TypeTags.BYTE_TAG:
+                    messageSerializer.setByteFieldValue((Integer) ballerinaValue);
+                    break;
+                case TypeTags.FLOAT_TAG:
+                    messageSerializer.setFloatFieldValue(ballerinaValue);
+                    break;
+                case TypeTags.STRING_TAG:
+                    messageSerializer.setStringFieldValue((BString) ballerinaValue);
+                    break;
+                case TypeTags.BOOLEAN_TAG:
+                    messageSerializer.setBooleanFieldValue((Boolean) ballerinaValue);
+                    break;
+                case TypeTags.DECIMAL_TAG:
+                    messageSerializer.setDecimalFieldValue((BDecimal) ballerinaValue);
+                    break;
+                case TypeTags.UNION_TAG:
+                    messageSerializer.setUnionFieldValue(ballerinaValue);
+                    break;
+                case TypeTags.ARRAY_TAG:
+                    messageSerializer.setArrayFieldValue((BArray) ballerinaValue);
+                    break;
+                case TypeTags.RECORD_TYPE_TAG:
+                    @SuppressWarnings("unchecked") BMap<BString, Object> recordValue
+                            = (BMap<BString, Object>) ballerinaValue;
+                    messageSerializer.setRecordFieldValue(recordValue);
+                    break;
+                case TypeTags.TUPLE_TAG:
+                    messageSerializer.setTupleFieldValue((BArray) ballerinaValue);
+                    break;
+                case TypeTags.TABLE_TAG:
+                    messageSerializer.setTableFieldValue((BTable<?, ?>) ballerinaValue);
+                    break;
+                case TypeTags.MAP_TAG:
+                    @SuppressWarnings("unchecked") BMap<BString, Object> mapValue
```

Comment:
Let's move `SuppressWarnings` to the line above the code.

Reply:
Updated